[12.0] split beesdoo_easy_my_coop

victor-champonnois opened the pull request:
## Description
Split beesdoo_easy_my_coop into:
- cooperator_eater rather then cooperator_eater_configuration
- cooperator_worker rather then cooperator_worker_configuration
- cooperator_info_session

Fix https://github.com/beescoop/Obeesdoo/issues/337
Fix https://github.com/beescoop/Obeesdoo/issues/276
Fix https://github.com/beescoop/Obeesdoo/issues/338
Fix https://github.com/beescoop/Obeesdoo/issues/339


**Misc**
- removed `test_max_eater_assignment_share_b`, I don't see a meaningful difference with `test_max_eater_assignment_share_a`
- removed `test_multiple_eater_assignement_share_a` and `test_parent_assignement_to_eater`: this is testing the ORM, not our code


---

### deprecated notes

Some questions I have:
- I put the `cooperator_type` field that was originally in `beesdoo_base` in `cooperator_worker_configuration`. I don't really understand the point of this field, but this seemed like a good place. @robinkeunen do you have an opinion ?
- `beesdoo_easy_my_coop` had an `auto-install` parameter, which means it installed itself automatically when its dependencies ("beesdoo_base", "beesdoo_shift", "cooperator", "cooperator_website", "partner_contact_birthdate") were installed. Should we keep this behavior for the splitted modules ? And if yes, for all of them ? (@vdewulf maybe you have an answer to that?) 

## Checklist before approval

- [ ] Tests are present (or not needed).
- [ ] Credits/copyright have been changed correctly.
- [ ] Change log snippet is present.
- [ ] (If a new module) Moving this to OCA has been considered.

victor-champonnois committed:
[REF] split beesdoo_easy_my_coop

[REF] Split beesdoo_emc -> cooperator_worker_configuration

[ADD] migration script

[UPD] Update cooperator_worker_configuration.pot

[REF] split beesdoo_emc -> cooperator_eater_configuration

[UPD] Update cooperator_eater_configuration.pot

[REF] beesdoo_emc -> cooperator_info_session

[UPD] Update cooperator_info_session.pot

[REM] auto-install

[IMP] cooperator_worker_configuration: add help message on is_worker

vdewulf commented:
@victor-champonnois About that:
`beesdoo_easy_my_coop had an auto-install parameter, which means it installed itself automatically when its dependencies ("beesdoo_base", "beesdoo_shift", "cooperator", "cooperator_website", "partner_contact_birthdate") were installed. Should we keep this behavior for the splitted modules ? And if yes, for all of them ? (@vdewulf maybe you have an answer to that?)`

I don't know what you will keep in the new splitted module, so I don't know if it would make sense to keep this behavior or not. Let's talk?

vdewulf commented:
Answer to beesdoo_easy_my_coop had an auto-install parameter, which means it installed itself automatically when its dependencies ("beesdoo_base", "beesdoo_shift", "cooperator", "cooperator_website", "partner_contact_birthdate") were installed. Should we keep this behavior for the splitted modules ? And if yes, for all of them ? (@vdewulf maybe you have an answer to that?)
Decision vdewulf/vchamponnois ===> no autoinstall

victor-champonnois commented:
status: waiting for beesdoo_base to deploy in test

robinkeunen commented:
I added a help message following discussion with @ferdymercury https://github.com/beescoop/Obeesdoo/issues/439#issuecomment-1196477827

robinkeunen commented:
@victor-champonnois @remytms @huguesdk j'ai mis à jour la description de la tâche, prêt pour une revue.

victor-champonnois commented on cooperator_worker/models/res_partner.py:
```diff
@@ -1,28 +1,30 @@
         readonly=True,
+        # override parent definition
         related="",
+        help="This field can only be set if Create Subscription "
+        "button was clicked on this user and payment was "
+        "validated, for a Share Type where Owner-can-work is checked ",
```
I have difficulty understanding the help message. Is it possible to make it clearer?

robinkeunen replied:
done

victor-champonnois reviewed (commented):
LGTM after a quick review, but I cannot approve because I created the PR ^^

robinkeunen commented:
@polchampion it's been deployed on bees-test, lenid-test, lepedalo-test and spp-test. Can you have a look ? Here are the descriptions of the new modules :

```
# cooperator_worker
- Add working and shopping information on a Share product
- Overwrite is_worker and can_shop based on share configuration.

# cooperator_eater
- Overwrite validation of subscription request to get eaters information
- Add constraint on eaters based on share configuration
- Add eater configuration on share product

# cooperator_info_session
Configuration for info session on website form for getting a cooperator share :

- info session description and fields on the online form
- add info session text description on the company
- set if the attending an info session is required on the company
```

This will need a release note and probably an update of the documentation.

huguesdk committed:
[FIX] exclude some modules from c.i. tests

set conflicting modules macavrac_base and polln_shift as rebel modules
to avoid conflicts during testing.

polchampion commented on cooperator_worker/README.rst:
```diff
@@ -0,0 +1,59 @@
+===============================
+Beescoop link with easy my coop
```
@victor-champonnois Will the strings "easy my coop" in readme be changed later ? Otherwise, best to update the readme now. (see ligne 22 as well)

robinkeunen replied:
merci 👍

polchampion commented on cooperator_worker/i18n/cooperator_worker_configuration.pot:
```diff
@@ -0,0 +1,63 @@
+#. module: cooperator_worker_configuration
+#: model:product.product,name:cooperator_worker_configuration.share_a
+#: model:product.template,name:cooperator_worker_configuration.share_a_product_template
+msgid "Share A"
```
@victor-champonnois @robinkeunen Je ne sais pas si les parts A, B et C de la bees sont toujours hardcodées ailleurs dans le code, mais elles sont ici dans le pot. Est-ce maintenant l'occasion de se débarrasser de ceci ?

robinkeunen replied:
Ce sont les traductions de données de démo. J'ai mis à jour les traductions, peux-tu y jeter un oeil ?

polchampion replied:
All good except one comment (see reply)

polchampion commented:
> # cooperator_worker
> - Overwrite is_worker and can_shop based on share configuration.

@victor-champonnois How is `can_shop` overwritten and based on what ? I thought there were no links between Cooperators and `can_shop`. Is `can_shop` introduced in `beesdoo_shift` or `cooperator` or both ?

> # cooperator_eater
> - Overwrite validation of subscription request to get eaters information

@victor-champonnois I don't undestand what this means

robinkeunen commented on .github/workflows/test.yml:
```diff
@@ -24,6 +24,15 @@ jobs:
       matrix:
         include:
           - container: ghcr.io/oca/oca-ci/py3.6-ocb12.0:latest
+            include: "macavrac_base"
+            makepot: "true"
+            name: test with OCB
+          - container: ghcr.io/oca/oca-ci/py3.6-ocb12.0:latest
+            include: "polln_shift"
+            makepot: "true"
+            name: test with OCB
+          - container: ghcr.io/oca/oca-ci/py3.6-ocb12.0:latest
+            exclude: "macavrac_base,polln_shift"
```
Est-ce que ceci est auto-généré depuis copier ? Sinon, on peut juste retirer les tests de polln et macavrac

huguesdk replied:
yes, copier asks for rebel module groups. they are tested separately. i don’t know whether it is possible to simply skip some modules. i’d rather have modules tested separately instead of not at all, because that would mean that we wouldn’t be aware if they got broken by some commit.

robinkeunen replied:
I think we can exclude them by removing the configuration from the test.yaml: there are no tests in those modules and we do not maintain them.

=> less time & less energy.

robinkeunen committed:
[IMP] cooperator_*: translations

robinkeunen commented:
@polchampion fixed the module descriptions.

polchampion commented:
Tested on `lepedalo-test`

> # cooperator_worker
- [x] Working and shopping information appear on a Share product
- [x] `is_worker` is checked based on share configuration
- [x] `can_shop` is checked based on share config unless overwritten by worker status.

> # cooperator_eater
- [x] Overwrite validation of subscription request to get eaters information  --> I don't understand what this means @robinkeunen 
- [x] Add constraint on eaters based on share configuration : Ok for effective cooperators
NO CONSTRAINT on eaters based on share configuration for candidate cooperators (as before) --> should the "eater" tab appear on candidate cooperators ?
- [x] Eater configuration appears on share product

> # cooperator_info_session
- [x] Configuration for info session appears on the company config (activate + required?)
- [x] Info session description and checkbox appears on website form

robinkeunen commented:
@polchampion DESCRIPRION.rst was changed to 

> - Set new cooperator as eater or worker based on the share configuration

polchampion commented on cooperator_worker/i18n/fr.po:
```diff
@@ -0,0 +1,119 @@
+#: model:ir.model.fields,field_description:cooperator_worker.field_res_partner__worker_store
+#: model:ir.model.fields,field_description:cooperator_worker.field_res_users__worker_store
+msgid "Force Worker"
+msgstr "Forcer le travailleur"
```
@robinkeunen  Ma suggestion pour la traduction: "Ajouter le candidat coopérateur au système des shifts". La fonctionnalité n'est pas affichée actuellement, cf [cette issue](https://github.com/beescoop/Obeesdoo/issues/374)

remytms commented:
/ocabot merge nobump
